src/tribol/interface/mfem_tribol.cpp

```diff
@@ -67,14 +67,14 @@ void registerMfemCouplingScheme( integer cs_id,
       {
          pressure_vdim = 1;
       }
-      else if (contact_model == TIED || contact_model == COULOMB)
+      else if (contact_model == TIED_NORMAL || contact_model == COULOMB)
```

Comment:
This should probably be `TIED_FULL` and `pressure_vdim = 1` for `TIED_NORMAL`

Reply:
The only issue here is that this part of the code is treating model and case options (the tied should be moved to check contact_case) that aren't supported or implemented. This conflates the coupling scheme checks with additional checks here that assign the pressure_vdim, exposing the potential for some inconsistency. On the other hand, if here we treat the whole kitchen sink, Tribol shouldn't get passed coupling scheme apply for unsupported/umimplemented combinations. With that in mind, perhaps I expand this on the case and model to include all enumerated options just for completeness?

Reply:
`TIED_NORMAL` only treats pressure in the normal direction (1 dimension) while `TIED_FULL` has a traction vector (in all spatial dimensions)?  It probably doesn't matter for now, since `pressure_vdim != 1` isn't used anywhere.  Maybe just set `pressure_vdim = 1` and add a note/TODO that this will need to be updated when we have more than a normal response.

More generally, we'll have to think how we want to handle/store this with tangential response and with full tied contact.  For tied contact, the components will likely align with the Cartesian axes, but potentially not for normal + tangential response.

Reply:
That is right. The normal component is only stored as a pressure, but the normal traction would simply be dotting that pressure with the common plane normal. The tangential component would likely be stored as a full traction. All of that is internal. These would still be integrated to give nodal force vectors, which is what is passed through the API.

I'll simplify this part of the code and put the TODO.

Reply:
Looing at this again, these are checks to set the pressure dim inside an if-statement when using Lagrange multipliers. Does this occur prior to calling coupling scheme init()? If so, we just check for Lagrange multipliers here, and not whether that enforcement is combined with an allowed method etc. Probably not a huge deal cause we just initialize and set some pressure stuff, but later when init() is called, we may find that the coupling scheme is not valid (e.g. Lagrange multipliers with common plane (not allowed) or with coulomb friction (not implemented).

Reply:
That's potentially true, but there isn't a way to create the pressure grid function after calling `CouplingScheme::init()`.  `CouplingScheme::init()` is trigged by calling `update()`, and when `update()` is called, all of the MFEM and Tribol data structures have already been created.  I think we'll have to leave this as is, pending a pretty major refactor.

Reply:
I think, in an ideal world, the MFEM API would not allow Tribol to get into an invalid state (i.e. `CouplingScheme::init()` should never fail), but I don't think we're there yet :smiley: